spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewriteDataFilesAction.java

```diff
@@ -1384,6 +1384,99 @@ public void testRewriteJobOrderFilesDesc() {
     Assert.assertNotEquals("Number of files order should not be ascending", actual, expected);
   }
 
+  @Test
+  public void testZOrderSortPartitionEvolution() {
+    int originalFiles = 20;
+    Table table = createTable(originalFiles);
+    shouldHaveLastCommitUnsorted(table, "c2");
+    shouldHaveFiles(table, originalFiles);
+
+    Stream.of(Expressions.bucket("c1", 2), Expressions.bucket("c2", 4))
```

Comment:
Any reason why we are adding this in two separate commits? Should repo with a single update correct?

Reply:
I wanted to have 3 diff partition specs to test